pkg/sources/generator/tickgen.go

```diff
@@ -217,42 +197,35 @@ func NewMemGen(
 		}
 	}
 
-	// attach a source publisher so the source can assign the watermarks.
-	genSrc.sourcePublishWM = genSrc.buildSourceWatermarkPublisher(publishWMStores)
+	// create a source watermark publisher
+	sourceWmPublisher := publish.NewSourcePublish(ctx, genSrc.pipelineName, genSrc.vertexName, publishWMStores,
```

Comment:
Same question as below, can we use `0` instead of Replica?

Reply:
Using '0' for sources like HTTP and Tickgen doesn't work, as we utilize the pod replica as the partition ID to publish the watermark. This approach helps us maintain coordination between the pods for the progression of watermarks.

Reply:
yes